core/src/main/scala/org/typelevel/scalacoptions/ScalaVersion.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.scalacoptions
+
+import org.typelevel.scalacoptions.internal.Parser
+
+import scala.Ordering.Implicits._
+import scala.collection.immutable
+
+abstract sealed case class ScalaVersion private (major: Long, minor: Long, patch: Long) {
+
+  def isBetween(addedVersion: ScalaVersion, removedVersion: ScalaVersion): Boolean =
+    this >= addedVersion && this < removedVersion
+}
+
+object ScalaVersion {
+  private def apply(major: Long, minor: Long, patch: Long): ScalaVersion =
+    new ScalaVersion(major, minor, patch) {}
+
+  lazy val knownVersions: immutable.SortedSet[ScalaVersion] = {
+    val bld = immutable.SortedSet.newBuilder[ScalaVersion]
+
+    def bldPatches(major: Long, minor: Long, lastPatch: Long): Unit =
+      for (patch <- 0L to lastPatch)
+        bld += ScalaVersion(major, minor, patch)
+
+    bldPatches(2, 11, 12)
+    bldPatches(2, 12, 17)
+    bldPatches(2, 13, 9)
+    bldPatches(3, 0, 2)
+    bldPatches(3, 1, 3)
+    bldPatches(3, 2, 0)
```

Comment:
Can we just leave it open ended? So unless a series is _known_ to be EOL it will accept not-yet-released versions?

Reply:
@armanbilge just to make sure I understand you correctly, 

you would prefer to avoid having scala-steward involved into that completely, wouldn't you?
I.e. rather than having scala-steward involved, you're suggesting to make `ScalaVersion` more flexible and supporting versions like "2.13+", "2.13.9+", etc?

Reply:
> you would prefer to avoid having scala-steward involved into that completely, wouldn't you?

Right, it's not that I care about Scala Steward specifically, but rather that we would have to release a new version of this library (and thus a new version of all the plugins that use it) for every new Scala version. 

> you're suggesting to make `ScalaVersion` more flexible and supporting versions like "2.13+", "2.13.9+", etc?

I don't think that's necessary. But I do think it should accept `2.13.20` today even though that's not yet released.